tiri/src/main.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 King's College London.
+// Created by the Software Development Team <http://soft-dev.org/>.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Tiri -- TIR interpreter.
+//!
+//! No effort has been made to make this fast.
+
+#[macro_use]
+extern crate log;
+
+use elf;
+use fallible_iterator::FallibleIterator;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::io::Cursor;
+use std::path::{Path, PathBuf};
+use ykpack::{BasicBlockIndex, Body, Decoder, DefId, Pack, Statement, StatementIndex, Terminator};
+
+/// The interpreter maintains a stack of these to keep track of calls.
+struct Frame<'p> {
```

Comment:
Why `'p` and not the more normal `'a` (or something else that makes sense in context)?

Reply:
`p` for pack, since that's where the stuff inside is borrowed from. But I don't feel strongly.

Reply:
OK.